libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -843,6 +848,8 @@ unsigned CompilerUtils::prepareMemoryStore(Type const& _type, bool _padToWordBou
 	else
 	{
 		solAssert(numBytes <= 32, "Memory store of more than 32 bytes requested.");
+		if (_type.category() == Type::Category::Bool)
```

Comment:
`cleanHigherOrderBits` currently works only for `IntegerType`.  I guess I should introduce a function of the same for `BoolType`.